### ISSUE TITLE
OKD on OpenStack docs refer to RHOSP only

### ISSUE THREAD

**Issue by woland7** [openshift/okd#511]
Hello, I'm not quite sure if this is the right place to ask, but I'm bugged with something. I have an OpenStack (Mitaka release) cluster, currently running OKD 3.11. I would like to updagre to OKD 4.6 but the documentation only says of RHOSP. So I'm not sure, can I OKD 4.6 onto Mitaka or a newer release or do I need RHOSP?

**Comment by adarshaj** [comment]
@woland7 - Documentation for OKD is currently in a state of flux, essentially it being an upstream for RHOCP, it reuses most of documentation with replacement of  OCP with OKD, but I do not understand why RHOSP (OpenStack) should matter? It is just your infrastructure provider. You can continue to maintain whatever OpenStack version you have currently and try to upgrade OKD. However, as you might know there's a lot of architectural changes from 3.x to 4.x, so it is probably better to start afresh with a 4.x release installation (within 4.x its supported . I am pretty sure given enough time and skill you can do an inplace upgrade, but do think about if its really worth the effort, it may be easier to install a parallel cluster with OKD 4.x and migrate the workload?

**Comment by LorbusChris** [comment]
Upgrades from 3.x to 4.x are not supported.

https://docs.okd.io/latest/installing/installing_openstack/installing-openstack-installer-custom.html does indeed lack mention of any upstream OpenStack versions. RHOSP 13 and 16.1 are supported in OCP, I don't know which upstream codenames those versions correspond to, though.

I will repurpose this issue to track this

/retitle OKD on OpenStack docs refer to RHOSP only

**Comment by woland7** [comment]
Yeah that's what I meant. Thank you for correcting me. And I made another mistake too, I misused the word _upgrade_. I had read you can't just upgrade from 3.x to 4.x, what I meant is that that I wanted to move to a fresh installation of 4.6 and I find there are some differences between RHOSP and upstream OpenStack (but I may be wrong again). Hope it can be clarified.

**Comment by woland7** [comment]
https://adam.younglogic.com/2015/11/translating-between-rdorhos-and-upstream-openstack-releases/comment-page-1/

Is this entry blog here reliable as to the matching between upstream OpenStack and RHOSP version?

**Comment by woland7** [comment]
After installing OKD 4.8 on OpenStack upstream Train, I can confirm those info.

RHOS Version 16 = upstream Train
RHOS Version 13 = upstream Queens

Those are the only two versions supported according to the docs.

Yet, I think the docs should be more clear and point to the upstream OpenStack versions, too. I'd be glad to update the docs, how should I get involved?

**Comment by LorbusChris** [comment]
@woland7 thank you for taking the initiative on this. Let's reopen this issue to track this.
I'll refer to @mburke5678 from the OpenShift docs team for the next steps here.

**Comment by woland7** [comment]
Ok thank you ;)

**Comment by mburke5678** [comment]
@LorbusChris @woland7 Sorry I missed this earlier. 

What is the ask here? 
* We need to swap out Red Hat OpenStack Platform (RHOSP) for OpenStack
* List the supported versions of OpenStack
** How does this change with OpenShift versions??
* List the differences between the two

**Comment by LorbusChris** [comment]
Per https://access.redhat.com/articles/4679401 the only supported versions of RHOSP in OCP 4.9 and 4.10 are RHOSP 16.1 and 16.2. So I think swapping out mentions of `RHOSP 16` with `OpenStack Train` for OKD would suffice here. Not sure if we need to explain the differences between the two in detail in these docs.

**Comment by mburke5678** [comment]
@LorbusChris The only problem that introduces is that this will become a maintenance issue we need to track with each release. I will ask the install writer if he can help us remember to do this.

**Comment by LorbusChris** [comment]
> 4.6 - 4.8 also supports Queen?

presumably, but those versions of OKD are not maintained anymore (OKD follows a rolling release model with only the latest version being supported, i.e. 4.10 at the moment.)

**Comment by mburke5678** [comment]
PR is merged.
